pandas/tests/frame/methods/test_shift.py

```diff
@@ -656,3 +656,45 @@ def test_shift_axis1_many_periods(self):
 
         shifted2 = df.shift(-6, axis=1, fill_value=None)
         tm.assert_frame_equal(shifted2, expected)
+
+    def test_shift_with_iterable(self):
```

Comment:
Can you split this up? This is testing many different things.

Reply:
Also, can you add a test for using multiple periods with `freq` and/or `fill_value` specified.

Reply:
I will.